doc/design/software-bridge-management.md

```diff
@@ -0,0 +1,495 @@
+	// Capabilities supported: (mac|ips), e.g. '{"mac": true}'
+	Capabilities string `json:"capabilities,omitempty"`
+	// IPAM configuration to be used for this network.
+	IPAM string `json:"ipam,omitempty"`
+	// name of the OVS bridge, if not set OVS will automatically select bridge
+	// based on VF PCI address
+	Bridge string `json:"bridge,omitempty"`
+	// VlanTag to assign for the OVS port
+	VlanTag uint `json:"vlan,omitempty"`
+	// Mtu for the OVS port
+	MTU int `json:"mtu",omitempty`
+	// Trunk configuration for the OVS port
+	Trunk []*OVSTrunkConfig `json:"trunk,omitempty"`
+	// OpenFlow port number in range 1 to 65,279
+	OfportRequest uint `json:"ofport_request,omitempty"`
+	// The type of interface on ovs.
+	InterfaceType string `json:"interface_type,omitempty"`
+	// Path to the configuration file on the host
+	ConfigurationPath string `json:"configuration_path,omitempty"`
+	// Paths to OVSDB socket
+	SocketFile string `json:"socket_file,omitempty"`
+	// Amount of retries for link state check
+	LinkStateCheckRetries int `json:"link_state_check_retries,omitempty"`
+	// Interval between retries for link state check
+	LinkStateCheckInterval int `json:"link_state_check_interval,omitempty"`
+	// MetaPluginsConfig configuration to be used in order to chain metaplugins
+	MetaPluginsConfig string `json:"metaPlugins,omitempty"`
+}
+
+// OVSTrunkConfig contains configuration for OVS trunk
+type OVSTrunkConfig struct {
+	MinID *uint `json:"minID,omitempty"`
+	MaxID *uint `json:"maxID,omitempty"`
+	ID    *uint `json:"id,omitempty"`
+}
+
+// OvsNetworkStatus defines the observed state of OvsNetwork
+type OvsNetworkStatus struct {
+}
+
+// OvsNetwork is the Schema for the ovsnetworks API
+type OvsNetwork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OvsNetworkSpec   `json:"spec,omitempty"`
+	Status OvsNetworkStatus `json:"status,omitempty"`
+}
+
+```
+
+#### BridgeNetwork CRD `new`
+
+```golang
+// BridgeNetworkSpec defines the desired state of BridgeNetwork
+type BridgeNetworkSpec struct {
+	// Namespace of the NetworkAttachmentDefinition custom resource
+	NetworkNamespace string `json:"networkNamespace,omitempty"`
+	// OVS Network device plugin endpoint resource name
+	ResourceName string `json:"resourceName"`
+	// Capabilities to be configured for this network.
+	// Capabilities supported: (mac|ips), e.g. '{"mac": true}'
+	Capabilities string `json:"capabilities,omitempty"`
+	// IPAM configuration to be used for this network.
+	IPAM string `json:"ipam,omitempty"`
+	// name of the Linux bridge, if not set will automatically select bridge
+	// based on VF PCI address
+	Bridge string `json:"bridge,omitempty"`
+	// VLAN ID for VF
+	Vlan int `json:"vlan,omitempty"`
+	// VLAN Trunk configuration
+	Trunk []BridgeTrunkConfig `json:"trunk,omitempty"`
+	// enable setting matching vlan tags on the bridge uplink interface, default is false
+	SetUplinkVlan bool `json:"setUplinkVlan,omitempty"`
```

Comment:
do we want something like this also for the ovsBridge API?

Reply:
OVSNetwork and BridgeNetwork CRDs are wrappers to create CNI args for ovs-cni and accelerated-bridge-cni. 
OVS-cni doesn't have such option as for now. It is not critical to have it in the OVS-cni, because by default ovs allows all vlans on a trunk port. Linux bridge require to set vlans on a trunk port explicitly.